profiling/paper_timing_tables/timing_comparison.py

```diff
@@ -0,0 +1,358 @@
+"""
+A script for comparing the timing of PINT and TEMPO/Tempo2 for standard fitting 
+cases. Recreates Tables 7, 8, and 9 from PINT paper.
+Requires: 
+    - TEMPO
+    - TEMPO2 
+*** THIS IS A PLACEHOLDER SCRIPT, final script will be in a notebook and cleaner/easier to work with. ***
+"""
+from pint import toa
+from pint import models
+from pint import fitter
+import os
+import sys
+import subprocess
+import timeit
+import datetime
+from astropy.table import Table
+from astropy.io import ascii
+
+MAXIT = 1  # number of iterations to time and average
+# number of TOAs run for each test
+ntoas_simple = [100, 1000, 10000, 100000]
+ntoas_complex = [5012, 10024, 25060]
+timfiles = [
+    "NGC6440E_fake100.tim",
+    "NGC6440E_fake1000.tim",
+    "NGC6440E_fake10000.tim",
+    "NGC6440E_fake100000.tim",
+]  # timfiles for simple model and individual functions
+timfiles2 = [
+    "J1910+1256_NANOGrav_12yv4.tim",
+    "J1910+1256_NANOGrav_12yv4_10k.tim",
+    "J1910+1256_NANOGrav_12yv4_25k.tim",
+]  # timfiles for complex model
+
+
+def pintrun(parfile, timfile, ptime_arr, pickle, fitter):
+    """ Runs and times pintempo 5 times and averages times, appending to a list. """
+    gls = ""
+    if fitter == "gls":
+        gls = " --gls"
+    usepickle = ""
+    if pickle:
+        usepickle = " --usepickle"
+    total = 0
+    for i in range(MAXIT):
+        start = timeit.default_timer()
+        subprocess.call(
```

Comment:
If you use `check_call` you will get an exception instead of wrong numbers if something goes wrong while running the subprocesses.